// docusaurus/lib/core/nav/SideNav.js
var _jsxFileName='website/core/nav/SideNav.js';var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var React=require('React');
var classNames=require('classnames');

var siteConfig=require(process.cwd()+'/siteConfig.js');var

SideNav=function(_React$Component){_inherits(SideNav,_React$Component);function SideNav(){_classCallCheck(this,SideNav);return _possibleConstructorReturn(this,(SideNav.__proto__||Object.getPrototypeOf(SideNav)).apply(this,arguments));}_createClass(SideNav,[{key:'render',value:function render()
{
return(
React.createElement('nav',{className:'toc',__source:{fileName:_jsxFileName,lineNumber:9}},
React.createElement('div',{className:'toggleNav',__source:{fileName:_jsxFileName,lineNumber:10}},
React.createElement('section',{className:'navWrapper wrapper',__source:{fileName:_jsxFileName,lineNumber:11}},
React.createElement('div',{className:'navBreadcrumb wrapper',__source:{fileName:_jsxFileName,lineNumber:12}},
React.createElement('div',{className:'navToggle',id:'navToggler',__source:{fileName:_jsxFileName,lineNumber:13}},React.createElement('i',{__source:{fileName:_jsxFileName,lineNumber:13}})),
React.createElement('h2',{__source:{fileName:_jsxFileName,lineNumber:14}},
React.createElement('i',{__source:{fileName:_jsxFileName,lineNumber:15}},'\u203A'),
React.createElement('span',{__source:{fileName:_jsxFileName,lineNumber:16}},this.props.current.category))),


React.createElement('div',{className:'navGroups',__source:{fileName:_jsxFileName,lineNumber:19}},
this.props.contents.map(this.renderCategory,this)))),



React.createElement('script',{
dangerouslySetInnerHTML:{
__html:'\n          var toggler = document.getElementById(\'navToggler\');\n          var nav = document.getElementById(\'docsNav\');\n          toggler.onclick = function() {\n            nav.classList.toggle(\'docsSliderActive\');\n          };\n        '},__source:{fileName:_jsxFileName,lineNumber:24}})));










}},{key:'renderCategory',value:function renderCategory(
category){
return(
React.createElement('div',{className:'navGroup navGroupActive',key:category.name,__source:{fileName:_jsxFileName,lineNumber:40}},
React.createElement('h3',{__source:{fileName:_jsxFileName,lineNumber:41}},this.getLocalizedCategoryString(category.name)),
React.createElement('ul',{__source:{fileName:_jsxFileName,lineNumber:42}},
category.links.map(this.renderItemLink,this))));



}},{key:'getLocalizedCategoryString',value:function getLocalizedCategoryString(
category){
var categoryString=
siteConfig[this.props.language]['localized-strings'][category];
if(typeof categoryString=='undefined'){
categoryString=category;
}
return categoryString;
}},{key:'getLocalizedString',value:function getLocalizedString(
metadata){
var localizedString='';
if(
typeof metadata.localized_id=='undefined'||
typeof siteConfig[this.props.language]=='undefined'||
typeof siteConfig[this.props.language]['localized-strings']=='undefined')
{
localizedString=metadata.title;
}else{
localizedString=
siteConfig[this.props.language]['localized-strings'][
metadata.localized_id];

}
return localizedString;
}},{key:'getLink',value:function getLink(
metadata){
if(metadata.permalink){
if(metadata.permalink.match(/^https?:/)){
return metadata.permalink;
}
return siteConfig.baseUrl+metadata.permalink+'#content';
}
if(metadata.path){
return siteConfig.baseUrl+'/blog/'+metadata.path;
}
return null;
}},{key:'renderItemLink',value:function renderItemLink(
link){
var itemClasses=classNames('navListItem',{
navListItemActive:link.id===this.props.current.id});

var linkClasses=classNames('navItem',{
navItemActive:link.id===this.props.current.id});

return(
React.createElement('li',{className:itemClasses,key:link.id,__source:{fileName:_jsxFileName,lineNumber:92}},
React.createElement('a',{className:linkClasses,href:this.getLink(link),__source:{fileName:_jsxFileName,lineNumber:93}},
this.getLocalizedString(link))));



}}]);return SideNav;}(React.Component);

SideNav.defaultProps={
contents:[]};

module.exports=SideNav;
