// docusaurus/lib/core/DocsLayout.js
const React = require('react');
const Container = require('./Container.js');
const Doc = require('./Doc.js');
const DocsSidebar = require('./DocsSidebar.js');
const Site = require('./Site.js');

class DocsLayout extends React.Component {
  render() {
    const metadata = this.props.metadata;
    const content = this.props.children;
    return React.createElement(
      Site,
      {
        config: this.props.config,
        className: 'sideNavVisible',
        section: 'docs',
        title: this.props.config[this.props.metadata.language]['localized-strings'][this.props.metadata.localized_id],
        description: content.trim().split('\n')[0],
        language: metadata.language
      },
      React.createElement(
        'div',
        { className: 'docMainWrapper wrapper' },
        React.createElement(DocsSidebar, { metadata: metadata }),
        React.createElement(
          Container,
          { className: 'mainContainer' },
          React.createElement(Doc, {
            content: content,
            config: this.props.config,
            source: metadata.source,
            title: this.props.config[this.props.metadata.language]['localized-strings'][this.props.metadata.localized_id],
            language: metadata.language
          }),
          React.createElement(
            'div',
            { className: 'docs-prevnext' },
            metadata.previous_id && React.createElement(
              'a',
              {
                className: 'docs-prev button',
                href: metadata.previous_id + '.html#content'
              },
              '\u2190',
              ' ',
              this.props.config[this.props.metadata.language]['localized-strings']['previous']
            ),
            metadata.next_id && React.createElement(
              'a',
              {
                className: 'docs-next button',
                href: metadata.next_id + '.html#content'
              },
              this.props.config[this.props.metadata.language]['localized-strings']['next'],
              ' ',
              '\u2192'
            )
          )
        )
      )
    );
  }
}
module.exports = DocsLayout;
