// docusaurus/lib/core/GridBlock.js
"use strict";

const React = require('react');
const classNames = require('classnames');

const Marked = require('./Marked.js');

class GridBlock extends React.Component {
  renderBlock(block) {
    const blockClasses = classNames('blockElement', this.props.className, {
      'alignCenter': this.props.align === 'center',
      'alignRight': this.props.align === 'right',
      'fourByGridBlock': this.props.layout === 'fourColumn',
      'imageAlignBottom': block.image && block.imageAlign === 'bottom',
      'imageAlignSide': block.image && (block.imageAlign === 'left' || block.imageAlign === 'right'),
      'imageAlignTop': block.image && block.imageAlign === 'top',
      'threeByGridBlock': this.props.layout === 'threeColumn',
      'twoByGridBlock': this.props.layout === 'twoColumn'
    });

    const topLeftImage = (block.imageAlign === 'top' || block.imageAlign === 'left') && this.renderBlockImage(block.image);

    const bottomRightImage = (block.imageAlign === 'bottom' || block.imageAlign === 'right') && this.renderBlockImage(block.image);

    return React.createElement(
      'div',
      { className: blockClasses, key: block.title },
      topLeftImage,
      React.createElement(
        'div',
        { className: 'blockContent' },
        this.renderBlockTitle(block.title),
        React.createElement(
          Marked,
          null,
          block.content
        )
      ),
      bottomRightImage
    );
  }

  renderBlockImage(image) {
    if (image) {
      return React.createElement(
        'div',
        { className: 'blockImage' },
        React.createElement('img', { src: image })
      );
    } else {
      return null;
    }
  }

  renderBlockTitle(title) {
    if (title) {
      return React.createElement(
        'h2',
        null,
        title
      );
    } else {
      return null;
    }
  }

  render() {
    return React.createElement(
      'div',
      { className: 'gridBlock' },
      this.props.contents.map(this.renderBlock, this)
    );
  }
}

GridBlock.defaultProps = {
  align: 'left',
  contents: [],
  imagealign: 'top',
  layout: 'twoColumn'
};

module.exports = GridBlock;
